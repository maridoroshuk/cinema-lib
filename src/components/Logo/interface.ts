export interface ILogo {
  size: TLogoSize;
}
