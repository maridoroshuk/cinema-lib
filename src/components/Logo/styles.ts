import styled from 'styled-components';

export const LogoStyled = styled.img<{
  width: number;
}>`
  width: 100%;
  max-width: ${({ width }) => width}px;
`;
