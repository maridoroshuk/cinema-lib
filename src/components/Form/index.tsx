import React from 'react';
import { Formik } from 'formik';

import { IFormProps } from './interface';
import { StyledForm } from './styles';

export const Form = <T extends {}>({
  children,
  initialValues,
  validate,
  onFormSubmit,
}: IFormProps<T>) => {
  return (
    <Formik
      initialValues={initialValues}
      validate={validate}
      onSubmit={onFormSubmit}
    >
      <StyledForm>{children}</StyledForm>
    </Formik>
  );
};
