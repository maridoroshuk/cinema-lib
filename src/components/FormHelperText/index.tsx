import React from 'react';

import { IFormHelperTextProps } from './interface';
import { Wrapper } from './styles';

export const FormHelperText = ({ message }: IFormHelperTextProps) => {
  return <Wrapper>{message}</Wrapper>;
};
