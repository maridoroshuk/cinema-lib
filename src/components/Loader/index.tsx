import React from 'react';
import ReactLoading from 'react-loading';
import { ILoaderProps } from './interface';
import { LoaderWrapper } from './styles';

export function Loader({ height, width, color }: ILoaderProps) {
  return (
    <LoaderWrapper>
      <ReactLoading type="spokes" color={color} height={height} width={width} />
    </LoaderWrapper>
  );
}
