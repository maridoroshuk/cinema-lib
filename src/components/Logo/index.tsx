import React from 'react';

import LogoImage from '@/assets/Logo.png';
import { getLogoSize } from '@/helpers/getLogoSize';

import { LogoStyled } from './styles';
import { ILogo } from './interface';

export const Logo = ({ size }: ILogo) => {
  const width = getLogoSize(size);
  return (
    <LogoStyled src={LogoImage} width={width} alt="company logo (Modsen)" />
  );
};
