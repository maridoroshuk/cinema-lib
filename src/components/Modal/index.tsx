import React from 'react';

import { IModal } from './interface';
import { Wrapper } from './styles';

export const Modal = ({ children }: IModal) => {
  return <Wrapper>{children}</Wrapper>;
};
