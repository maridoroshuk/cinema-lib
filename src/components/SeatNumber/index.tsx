import React from 'react';

import { ISeatProps } from '@/components/SeatNumber/interface';
import { StyledSeat } from '@/components/SeatNumber/styles';

export const SeatNumber = ({
  children,
  backgroundColor,
  index,
}: Partial<ISeatProps>) => {
  return (
    <StyledSeat backgroundColor={backgroundColor} seatIndex={index}>
      {children}
    </StyledSeat>
  );
};
