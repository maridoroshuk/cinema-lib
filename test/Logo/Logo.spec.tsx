import React from 'react';
import { render } from '@testing-library/react';

import { Logo } from '@/components/Logo';

import 'jest-styled-components';

import { setupTestProvider } from '../setupTests';

describe('Logo', () => {
  it('renders small logo', () => {
    const { container } = render(setupTestProvider(<Logo size="small" />));
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyle({
      'max-width': '125px',
    });
  });
  it('renders medium logo', () => {
    const { container } = render(setupTestProvider(<Logo size="medium" />));
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyle({
      'max-width': '175px',
    });
  });
  it('renders large logo', () => {
    const { container } = render(setupTestProvider(<Logo size="large" />));
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyle({
      'max-width': '255px',
    });
  });
});
