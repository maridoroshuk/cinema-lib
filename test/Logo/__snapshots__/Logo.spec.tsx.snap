// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders large logo 1`] = `
.c0 {
  width: 100%;
  max-width: 255px;
}

<img
  alt="company logo (Modsen)"
  class="c0"
  src=""
  width="255"
/>
`;

exports[`Logo renders medium logo 1`] = `
.c0 {
  width: 100%;
  max-width: 175px;
}

<img
  alt="company logo (Modsen)"
  class="c0"
  src=""
  width="175"
/>
`;

exports[`Logo renders small logo 1`] = `
.c0 {
  width: 100%;
  max-width: 125px;
}

<img
  alt="company logo (Modsen)"
  class="c0"
  src=""
  width="125"
/>
`;
